metrics/influx_test.go

```diff
@@ -0,0 +1,109 @@
+package metrics
+
+import (
+	"fmt"
+	"github.com/containous/traefik/types"
+	"net"
+	"net/http"
+	"regexp"
+	"strconv"
+	"testing"
+	"time"
+)
+
+var (
+	testAddr = ":8089"
+	addr     *string
+	listener *net.UDPConn
+	Timeout  time.Duration
+)
+
+func TestInflux(t *testing.T) {
+	SetAddr(testAddr)
+
+	Timeout = 5 * time.Second
+
+	influxRegistry := RegisterInflux(&types.Influx{Address: "localhost:8089", PushInterval: "1s"})
+	defer StopInflux()
+
+	if !influxRegistry.IsEnabled() {
+		t.Errorf("InfluxRegistry should return true for IsEnabled()")
+	}
+
+	expected := []string{
+		// We are only validating counts, as it is nearly impossible to validate latency, since it varies every run
+		"(requests.total,code=200,method=GET,service=test count=1) [0-9]{19}",
+		"(requests.total,code=404,method=GET,service=test count=1) [0-9]{19}",
+		"(request.duration,code=200,method=GET,service=test p50=10000,p90=10000,p95=10000,p99=10000) [0-9]{19}",
+	}
+
+	got := getMessage(t, func() {
+		influxRegistry.ReqsCounter().With("service", "test", "code", strconv.Itoa(http.StatusOK), "method", http.MethodGet).Add(1)
+		influxRegistry.ReqsCounter().With("service", "test", "code", strconv.Itoa(http.StatusNotFound), "method", http.MethodGet).Add(1)
+		influxRegistry.ReqDurationHistogram().With("service", "test", "code", strconv.Itoa(http.StatusOK)).Observe(10000)
+	})
+
+	if err := extractAndMatchMessage(t, expected, got); err != nil {
+		t.Errorf(err.Error())
+	}
+
+}
+
+type fn func()
+
+func getMessage(t *testing.T, body fn) string {
```

Comment:
I can see that you have duplicated some code from [go-udp-testing](https://github.com/stvp/go-udp-testing/blob/master/udp.go#L43). We are not able to accept your PR like that. 
You have opened an [issue](https://github.com/stvp/go-udp-testing/issues/5) on go-udp-testing. Could you try do open a PR please ?

Reply:
Related PR https://github.com/stvp/go-udp-testing/pull/6